### PR TITLE
Added integration tests for paybill

### PR DESCRIPTION
**Pull request by truthfool**
- Cucumber integration tests for paybill
- Cucumber tests for bpmn upload,validation endpoint and settlement endpoint for paygops and roster
@avikganguly01 @fynmanoj

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -6,14 +6,16 @@ plugins {
 
 group = 'org.mifos'
 version = '0.0.1-SNAPSHOT'
-sourceCompatibility = '17'
+sourceCompatibility = '11'
```
Comment (danishjamal104) [inline build.gradle:9]: This change we made for local only, avoid pushing it to VCS.

File: src/test/java/org/mifos/integrationtest/common/dto/paybill/BPMNUploadRequestDTO.java [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package org.mifos.integrationtest.common.dto.paybill;
```
Comment (danishjamal104) [inline src/test/java/org/mifos/integrationtest/common/dto/paybill/BPMNUploadRequestDTO.java:1]: Are we using this file anywhere??

File: src/test/java/org/mifos/integrationtest/common/dto/paybill/PaybillResponseDTO.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
+package org.mifos.integrationtest.common.dto.paybill;
+
+public class PaybillResponseDTO {
```
Comment (danishjamal104) [inline src/test/java/org/mifos/integrationtest/common/dto/paybill/PaybillResponseDTO.java:3]: Use `lambok`.

File: src/test/java/org/mifos/integrationtest/cucumber/PaybillApiStepDef.java [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import io.restassured.RestAssured;
+import io.restassured.builder.ResponseSpecBuilder;
+import io.restassured.http.ContentType;
+import io.restassured.specification.RequestSpecification;
+import org.mifos.integrationtest.common.Utils;
+import org.mifos.integrationtest.common.dto.paybill.PayBillRequestDTO;
+import org.mifos.integrationtest.common.dto.paybill.PaybillResponseDTO;
+
```
Comment (danishjamal104) [inline src/test/java/org/mifos/integrationtest/cucumber/PaybillApiStepDef.java:15]: we can avoid empty lines b/w imports.

File: src/test/java/org/mifos/integrationtest/cucumber/PaybillStepDef.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    public String amsName;
+    @Autowired
+    PaybillConfig paybillConfig;
+    @Autowired
+    ObjectMapper objectMapper;
+    String paybillBpmnName;
+    String response;
+    Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    public String getAbsoluteFilePathToResource() {
```
Comment (danishjamal104) [inline src/test/java/org/mifos/integrationtest/cucumber/PaybillStepDef.java:19]: the generic version of this method is already implemented in  `Utils.java` file in package `org.mifos.integrationtest.common`, please use that.

**Review by danishjamal104 — CHANGES_REQUESTED** [review]

File: src/test/java/org/mifos/integrationtest/common/dto/paybill/BPMNUploadRequestDTO.java (hunk 2 above)
Comment (truthfool) [reply src/test/java/org/mifos/integrationtest/common/dto/paybill/BPMNUploadRequestDTO.java:1]: Removed it

File: src/test/java/org/mifos/integrationtest/common/dto/paybill/PayBillRequestDTO.java [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+//        "TransTime":"20191122063845",
+//        "TransAmount":"10",
+//        "BusinessShortCode":"24322607",
+//        "BillRefNumber":"24322607",
+//        "InvoiceNumber":"",
+//        "OrgAccountBalance":"49197.00",
+//        "ThirdPartyTransID":"",
+//        "MSISDN":"254797668592",
+//        "FirstName":"John"
+//}
```
[the remark this answers is not in the document]
Comment (truthfool) [reply src/test/java/org/mifos/integrationtest/common/dto/paybill/PayBillRequestDTO.java:14]: Ok I will add the annotations for it.

File: src/test/java/org/mifos/integrationtest/cucumber/PaybillStepDef.java (hunk 5 above)
Comment (truthfool) [reply src/test/java/org/mifos/integrationtest/cucumber/PaybillStepDef.java:19]: It is not there.

File: cucumber.json [hunk 7]
```diff
@@ -0,0 +1 @@
...k api with expected status code of 200(file:///Users/ishan/fynarfin/phee/ph-ee-connector-integration-test/src/test/java/resources/paybill.feature:15)\n","duration":3365000,"status":"failed"},"line":15,"name":"I call the mpesa-connector validate webhook api with expected status code of 200","match":{"arguments":[{"val":"200","offset":77}],"location":"org.mifos.integrationtest.cucumber.PaybillApiStepDef.callMpesaConnector(int)"},"keyword":"And "},{"result":{"duration":54000,"status":"skipped"},"line":16,"name":"The response contains reconciled field as \"true\"","match":{"arguments":[{"val":"\"true\"","offset":42}],"location":"org.mifos.integrationtest.cucumber.PaybillApiStepDef.checkValidationResponse(java.lang.String)"},"keyword":"When "},{"result":{"duration":34000,"status":"skipped"},"line":17,"name":"I call the confirmation webhook API with expected status of 200","match":{"arguments":[{"val":"200","offset":60}],"location":"org.mifos.integrationtest.cucumber.PaybillApiStepDef.callConfirmationWebhook(int)"},"keyword":"Then "}]}],"name":"Paybill API Tests","description":"","id":"paybill-api-tests","keyword":"Feature","uri":"file:src/test/java/resources/paybill.feature","tags":[]}]
```
[the remark this answers is not in the document]
Comment (truthfool) [reply cucumber.json:1]: ok

File: src/test/java/org/mifos/integrationtest/cucumber/PaybillStepDef.java (hunk 5 above)
Comment (danishjamal104) [reply src/test/java/org/mifos/integrationtest/cucumber/PaybillStepDef.java:19]: that PR is not merged my manoj. @fynmanoj can you plzz merge my PR?? 
https://github.com/openMF/ph-ee-connector-integration-test/pull/3

Comment (truthfool) [reply src/test/java/org/mifos/integrationtest/cucumber/PaybillStepDef.java:19]: ok

File: src/main/resources/application.yaml [hunk 8]
```diff
@@ -7,3 +7,13 @@ operations-app:
...
     batch-details: "api/v1/batch/detail"
     batch-transaction: "/api/v1/batch/transactions"
+paybill:
+  mpesa-connector:
+    contactpoint: https://mpesa.sandbox.fynarfin.io
+    endpoints:
+      validation: "/validation"
+      settlement: "/confirmation"
+  zeebe-operations:
+    contactpoint: "https://zeebeops.sandbox.fynarfin.io"
```
Comment (fynmanoj) [inline src/main/resources/application.yaml:17]: Give dummy URLs here and and helm variables accordingly

File: src/test/java/org/mifos/integrationtest/common/Utils.java [hunk 9]
```diff
@@ -3,11 +3,11 @@
 import io.restassured.RestAssured;
 import io.restassured.builder.RequestSpecBuilder;
 import io.restassured.specification.RequestSpecification;
-
 public class Utils {
 
     public static final String TENANT_PARAM_NAME = "Platform-TenantId";
     public static final String DEFAULT_TENANT = "ibank-india";
+    public static final String AMS_NAME = "amsName";
```
Comment (fynmanoj) [inline src/test/java/org/mifos/integrationtest/common/Utils.java:10]: In the above two lines, parameterize the hard-coded values

File: src/test/java/resources/paybill_workflow.bpmn [hunk 10]
```diff
@@ -0,0 +1,219 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (fynmanoj) [inline src/test/java/resources/paybill_workflow.bpmn:1]: Why are we adding the BPMN here?

**Review by fynmanoj — CHANGES_REQUESTED** [review]

**Pull request closed without merging**